modules/op-installing-tekton-hub-with-login-and-rating.adoc

```diff
@@ -0,0 +1,111 @@
+
+** For a link:https://docs.github.com/en/developers/apps/creating-an-oauth-app[GitHub OAuth application], set the Homepage URL and the Authorization callback URL as `<auth-route>`.
+
+** For a link:https://docs.gitlab.com/ee/integration/oauth_provider.html#user-owned-applications[GitLab OAuth application], set the `REDIRECT_URI` as `<auth-route>/auth/gitlab/callback`.
+
+** For a link:https://support.atlassian.com/bitbucket-cloud/docs/use-oauth-on-bitbucket-cloud[BitBucket OAuth application], set the `Callback URL` as `<auth-route>`.
+
+. Edit the `<tekton_hub_repository>/config/02-api/20-api-secret.yaml` file of your cloned repository to include the {tekton-hub} API secrets:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: tekton-hub-api
+  namespace: openshift-pipelines
+type: Opaque
+stringData:
+  GH_CLIENT_ID: <1>
+  GH_CLIENT_SECRET: <2>
+  GL_CLIENT_ID: <3>
+  GL_CLIENT_SECRET: <4>
+  BB_CLIENT_ID: <5>
+  BB_CLIENT_SECRET: <6>
+  JWT_SIGNING_KEY: <7>
+  ACCESS_JWT_EXPIRES_IN: <8>
+  REFRESH_JWT_EXPIRES_IN: <9>
+  AUTH_BASE_URL: <10>
+  GHE_URL: <11>
+  GLE_URL: <12>
+----
+<1> The Client ID from the GitHub OAuth application.
+<2> The Client Secret from the GitHub OAuth application.
+<3> The Client ID from the GitLab OAuth application.
+<4> The Client Secret from the GitLab OAuth application.
+<5> The Client ID from the BitBucket OAuth application.
+<6> The Client Secret from the BitBucket OAuth application.
+<7> A long, random string used to sign the JSON Web Token (JWT) created for users.
+<8> Add the time limit after which the access token expires. For example, `1m`, where m denotes minutes. The supported units of time are seconds (`s`), minutes (`m`), hours (`h`), days (`d`), and weeks (`w`).
+<9> Add the time limit after which the refresh token expires. For example, `1m`, where `m` denotes minutes. The supported units of time are seconds (`s`), minutes (`m`), hours (`h`), days (`d`), and weeks (`w`). Ensure that the expiry time set for token refresh is greater than the expiry time set for token access.
+<10> Route URL for the OAuth application.
+<11> GitHub Enterprise URL, if you are authenticating using GitHub Enterprise. Do not provide the URL to the catalog as a value for this field.
+<12> GitLab Enterprise URL, if you are authenticating using GitLab Enterprise. Do not provide the URL to the catalog as a value for this field.
++
+[NOTE]
+====
+You can delete the unused fields for the Git repository hosting service providers that are irrelevant to your deployment.
+====
+
+. Create a `TektonHub` CR similar to the following example.
++
+[source,yaml]
+----
+apiVersion: operator.tekton.dev/v1alpha1
+kind: TektonHub
+metadata:
+  name: hub
+spec:
+  targetNamespace: openshift-pipelines <1>
+  api:
+    hubConfigUrl: https://raw.githubusercontent.com/tektoncd/hub/main/config.yaml <2>
```

Comment:
Are we allowed to reference upstream repositories?

Reply:
This has been discussed and debated thoroughly. Till the time Tekton Hub repository is brought under a fully Red Hat controlled organization on GitHub, this is the best we can do.

Slack discussion thread: https://coreos.slack.com/archives/CG5GV6CJD/p1666353260562669